static_frame/core/index.py

```diff
@@ -382,6 +383,16 @@ def copy(self: I) -> I:
         '''
         return self.__copy__() #type: ignore
 
+    def __sizeof__(self: I) -> I:
+        # TODO: Could labels and positions be is-equal?
```

Comment:
Yes they can, in which case I don't think we should double count.

Also, since `self.positions` is always going to be a slice of a global numpy array, I'm not sure `sys.getsizeof` will be accurate. Perhaps we need to use `self.positions.nbytes` instead?

Reply:
Working on a function that will prevent double-counting objects with the same id.

`sys.getsizeof` will return the size of the reference numpy array. Since the index is just storing a reference, I think that only the size of the reference should be counted toward the index's size.

To get the size of the global numpy array, maybe we should allow for a call to getsizeof on the PositionsAllocator?

Reply:
Got double-count prevention working!